core/dbt/parser/unit_tests.py

```diff
@@ -105,44 +106,66 @@ def parse_unit_test_case(self, test_case: UnitTestDefinition):
               - {id: 2, b: 2}
         """
         # Add the model "input" nodes, consisting of all referenced models in the unit test.
-        # This creates a model for every input in every test, so there may be multiple
-        # input models substituting for the same input ref'd model.
+        # This creates an ephemeral model for every input in every test, so there may be multiple
+        # input models substituting for the same input ref'd model. Note that since these are
+        # always "ephemeral" they just wrap the tested_node SQL in additional CTEs. No actual table
+        # or view is created.
         for given in test_case.given:
             # extract the original_input_node from the ref in the "input" key of the given list
             original_input_node = self._get_original_input_node(given.input, tested_node)
 
-            original_input_node_columns = None
-            if (
-                original_input_node.resource_type == NodeType.Model
-                and original_input_node.config.contract.enforced
-            ):
-                original_input_node_columns = {
-                    column.name: column.data_type for column in original_input_node.columns
-                }
-
-            # TODO: include package_name?
             input_name = f"{unit_test_node.name}__{original_input_node.name}"
-            input_unique_id = f"model.{package_name}.{input_name}"
-            input_node = ModelNode(
-                raw_code=self._build_fixture_raw_code(
-                    given.get_rows(
-                        self.root_project.project_root, self.root_project.fixture_paths
+            common_fields = {
+                "resource_type": NodeType.Model,
+                "package_name": package_name,
+                "original_file_path": original_input_node.original_file_path,
+                "unique_id": f"model.{package_name}.{input_name}",
+                "config": ModelConfig(materialized="ephemeral"),
+                "database": original_input_node.database,
+                "alias": original_input_node.identifier,
+                "schema": original_input_node.schema,
+                "fqn": original_input_node.fqn,
+                "checksum": FileHash.empty(),
+            }
+
+            if original_input_node.resource_type == NodeType.Model:
+                original_input_node_columns = None
+                if original_input_node.config.contract.enforced:
+                    original_input_node_columns = {
+                        column.name: column.data_type for column in original_input_node.columns
+                    }
```

Comment:
ah, just seeing this. I guess we're not just removing it as above here: https://github.com/dbt-labs/dbt-core/pull/9059/files#r1391606187

I think at this point with what we've landed on from https://github.com/dbt-labs/dbt-core/issues/8757#issuecomment-1796297194, we could actually remove this attempt to infer columns from here (perhaps doing the cleanup in macros in a follow-on issue). That way we can consolidate `raw_code` to `common_fields` as well.